app/helpers/samples_helper.rb

```diff
@@ -122,8 +122,30 @@ def samples_pipeline_run_info(samples)
     pipeline_run_info
   end
 
-  def samples_info(samples)
-    { final_result: samples_output_data(samples),
-      pipeline_run_info: samples_pipeline_run_info(samples) }
+  def filter_samples(samples, query)
+    filtered = []
+    samples.each do |output|
+      pipeline_run_status = output.pipeline_runs.first ? output.pipeline_runs.first.job_status : nil
+      if query == 'UPLOADING'
+        filtered.push(output) unless pipeline_run_status
+      elsif pipeline_run_status == query
+        filtered.push(output)
```

Comment:
change to 

samples = samples.joins("INNER JOIN pipeline_runs ON pipeline_runs.sample_id and samples.id").where(status: 'checked').where("pipeline_runs.id in (select max(id)  from pipeline_runs group by sample_id)").where("pipeline_runs.job_status = #{query}")

Reply:
returns inaccurate samples